src/internal_events/nginx_metrics.rs

```diff
@@ -6,11 +6,34 @@ use vector_core::internal_event::InternalEvent;
 
 use crate::sources::nginx_metrics::parser::ParseError;
 
+#[derive(Debug)]
+pub struct NginxMetricsBytesReceived<'a> {
+    pub byte_size: usize,
+    pub protocol: &'a str,
+    pub endpoint: &'a str,
+}
+
+impl InternalEvent for NginxMetricsBytesReceived<'_> {
+    fn emit(self) {
+        trace!(
+            message = "Bytes received.",
+            byte_size = %self.byte_size,
+            protocol = %self.protocol,
+            endpoint = %self.endpoint,
+        );
+        counter!(
+            "component_received_bytes_total", self.byte_size as u64,
+            "protocol" => self.protocol.to_owned(),
+            "endpoint" => self.endpoint.to_owned(),
+        );
+    }
+}
```

Comment:
This appears to be identical to `PostgresqlMetricsBytesReceived` and `MongodbMetricsBytesReceived`. Could we be using generic for them?

Reply:
So I did have this thought.

I guess this could be `BytesReceived` and we just force needing to provide an endpoint and protocol for that?

Mostly I've struggled figuring out what the most basic versions of bytes received, sent, etc, should look like tag-wise, and sort of went for the most isolated approach where sources define their own versions.

Reply:
Did a bunch of clean in [6261569](https://github.com/vectordotdev/vector/pull/12572/commits/6261569749f84f463ea7ad9d0f425ab211886607) that addresses this.